allocator.go

```diff
@@ -6,45 +6,45 @@ import (
 	"time"
 )
 
-// NodeID is a 64-bit identifier for nodes.
-type NodeID int64
+// nodeId is a 64-bit identifier for nodes.
+type nodeId int64
 
-// RangeID is a 64-bit identifier for ranges.
-type RangeID int64
+// rangeId is a 64-bit identifier for ranges.
+type rangeId int64
 
-// Resource is 32-bit identifier for resources -> {Disk, Qps}
-type Resource int
+// resource is 32-bit identifier for resources -> {disk, qps}
+type resource int
 
 // Allocation is the return type of our allocator.
 // It models mappings of RangeIDs to a list of NodeIDs.
-type Allocation map[RangeID][]NodeID
+type Allocation map[int64][]int64
```

Comment:
Why did we make this change?

Reply:
@smcheema isnt it better if we have type Allocation map[rangeID][]nodeId its much easier to follow. We already defined node id and range id as a type why not use it here

Reply:
I thought it didn't add much to the caller given the newer interface as it abstracts away so much already. My rationale was that if I'm using the allocator, and not building the `nodes/ranges` myself, I don't need to interact with however `IDs` are built under the hood. Given that they're defined as numeric types, previously each caller would need to call `RangeID(someNumber)/NodeID(someNumber)`, pass it to us and then convert back into some primitive once receiving a result, I thought this could instead belong in the constructor, and the need to expose `RangeID/NodeID` wasn't that necessary as it's just another cast callers would end up doing. I don't think any of the above are huge benefits, and I leave it up to you and @AlirezaDX to see which approach we should implement, both seem good to me.